src/core/mesh/qgsmeshlayer.h

```diff
@@ -289,6 +291,37 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer
       */
     QgsMesh3dDataBlock dataset3dValue( const QgsMeshDatasetIndex &index, const QgsPointXY &point ) const;
 
+    /**
+      * Returns the value of 1D mesh dataset defined on edge that are in the rectangle and the closest of the center of this rectangle
+      *
+      * \note It uses previously cached and indexed triangular mesh
+      * and so if the layer has not been rendered previously
+      * (e.g. when used in a script) it returns NaN value
+      *
+      * \param index dataset index specifying group and dataset to extract value from
+      * \param rectangle the area where to search the edge, closest edge from center point will be used
+      * \returns interpolated value at the projected point. Returns NaN values for values
+      * outside the mesh layer and in case triangular mesh was not previously used for rendering
+      *
+      * \since QGIS 3.14
+      */
+    QgsMeshDatasetValue dataset1DValue( const QgsMeshDatasetIndex &index, const  QgsRectangle &rectangle ) const;
+
+    /**
+      * Returns position in map coordinates of the closest vertex of pointCenter and in the radius searchRadius
+      *
+      * \note It uses previously cached and indexed triangular mesh
+      * and so if the layer has not been rendered previously
+      * (e.g. when used in a script) it returns empty point
+      *
+      * \param pointCenter the center point of the search area, in map coordinates
+      * \param searchRadius the radius of the search area in map unit
+      * \returns coordinate point of the vertex in map coordinates
+      *
+      * \since QGIS 3.14
+      */
+    QgsPointXY snapOnVertex( const QgsPointXY &pointCenter, double searchRadius ) const;
```

Comment:
This function is not used anywhere , please remove

Reply:
Its aim is to be used with crayfish for tracing a line along which will be pot on graph the dataset value

Reply:
all plots in crayfish usually just use datasetValue (/1d/3d) directoly? maybe I do not understand which plot?

Reply:
The idea is to make longitudinal plot for 1D with the same principle that cross section plot. But to allow the line to fit on the 1D network, the user could click near vertices and the line vertices for plot will fir with network vertices.

Reply:
ins't it possible to do directly in the python code?